dotnet/samples/KernelSyntaxExamples/Example95_GeminiTextGeneration.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Connectors.GoogleVertexAI;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Examples;
+
+public sealed class Example95_GeminiTextGeneration : BaseTest
+{
+    [Fact]
+    public async Task GoogleAIAsync()
+    {
+        this.WriteLine("============= Google AI - Gemini Text Generation =============");
+
+        string geminiApiKey = TestConfiguration.GoogleAI.ApiKey;
+        string geminiModelId = TestConfiguration.GoogleAI.Gemini.ModelId;
+
+        if (geminiApiKey is null || geminiModelId is null)
+        {
+            this.WriteLine("Gemini google ai credentials not found. Skipping example.");
+            return;
+        }
+
+        Kernel kernel = Kernel.CreateBuilder()
+            .AddGoogleAIGeminiTextGeneration(
+                modelId: geminiModelId,
+                apiKey: geminiApiKey)
+            .Build();
+
+        await RunSampleAsync(kernel);
+    }
+
+    [Fact]
+    public async Task VertexAIAsync()
+    {
+        this.WriteLine("============= Vertex AI - Gemini Text Generation =============");
+
+        string geminiApiKey = TestConfiguration.VertexAI.BearerKey;
+        string geminiModelId = TestConfiguration.VertexAI.Gemini.ModelId;
+        string geminiLocation = TestConfiguration.VertexAI.Location;
+        string geminiProject = TestConfiguration.VertexAI.ProjectId;
+
+        if (geminiApiKey is null || geminiModelId is null || geminiLocation is null || geminiProject is null)
+        {
+            this.WriteLine("Gemini vertex ai credentials not found. Skipping example.");
+            return;
+        }
+
+        Kernel kernel = Kernel.CreateBuilder()
+            .AddVertexAIGeminiTextGeneration(
```

Comment:
What's the difference between GoogleAIGemini and VertexAIGemini?

Reply:
Both GoogleAI and Vertex supports gemini models.
These are different endpoints. GoogleAI is simple platform for customer. But VertexAI has more capabilities.
There are slight differences between them.

Instead, embeddings endpoints vary more.

More details are here https://cloud.google.com/vertex-ai/generative-ai/docs/migrate/migrate-google-ai#google-ai